src/charts/bar.js

```diff
@@ -326,8 +325,17 @@ export default function module() {
                 }));
         }
 
-        colorMap = (item) =>
-            colorList.filter(({ name }) => name === item)[0].color;
+        nameToColorMap =
+            nameToColorMap ||
+            data
+                .map((d) => d)
+                .reduce(
+                    (acc, { name }, i) => ({
+                        ...acc,
+                        [name]: colorSchema[i % colorSchema.length],
```

Comment:
what does the modulo operator do in this case?

Reply:
It applies one color from the color schema to each category, starting again when it reaches the max number of elements, so for a colorSchema of 2 colors:
- first element, i = 0, color colorSchema[0]
- second element, i = 1, color colorSchema[1]
- third element, i = 2, color colorSchema[0]
and so on